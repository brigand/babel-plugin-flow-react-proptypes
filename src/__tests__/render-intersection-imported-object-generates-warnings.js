const content = `
import type {TestType} from './test-types.js.transpiled';

import React from 'react';
import renderer from 'react-test-renderer';

type FooProps = TestType & {
  foo: string,
  bar: number,
};

class C extends React.Component {
    props: FooProps
    
    render() { return <div/> };
}
  const tree1 = renderer.create(
    <C/>
  );`;

const utils = require('./lib/render-component');

it('intersection-with-imported-object-generates-warnings', () => {
  const path = require('path');
  const errorsSeen = utils.getConsoleErrorsForComponent(content, [
    [path.join(__dirname, 'lib', 'test-types.js.pre-transpile'),
      path.join(__dirname, 'lib', 'test-types.js.transpiled')]
  ]);
  expect(errorsSeen).toMatchSnapshot();
});

