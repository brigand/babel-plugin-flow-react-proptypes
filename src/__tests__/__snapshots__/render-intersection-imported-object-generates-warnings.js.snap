exports[`test intersection-with-imported-object-generates-warnings 1`] = `
Array [
  Array [
    "Warning: Failed prop type: The prop \`baz\` is marked as required in \`C\`, but its value is \`undefined\`.
    in C",
  ],
  Array [
    "Warning: Failed prop type: The prop \`foo\` is marked as required in \`C\`, but its value is \`undefined\`.
    in C",
  ],
  Array [
    "Warning: Failed prop type: The prop \`bar\` is marked as required in \`C\`, but its value is \`undefined\`.
    in C",
  ],
]
`;
